### ISSUE TITLE
[java] `UselessParentheses` around `switch` expression

### ISSUE THREAD

**Issue by vbrandl** [pmd/pmd#6123]
**Affects PMD Version:** 6.55.0, 7.13.0

**Rule:** [UselessParentheses](https://docs.pmd-code.org/pmd-doc-7.13.0/pmd_rules_java_codestyle.html#uselessparentheses)

**Description:**

When chaining a method call directly after a switch expression, the expression has to be wrapped into parentheses. Removing the parentheses is a syntax error (at least in Java 17.0.16). PMD detects this as a `UselessParentheses` rule violation.

**Code Sample demonstrating the issue:**

```java
public class Main {
  public enum E {
    A, B;
  }

  public static String foo(final E e) {
    return (switch (e) {
      case A -> "bar";
      case B -> "baz";
    }).toUpperCase();
  }
}
```

**Expected outcome:**

PMD shouldn't detect the parentheses around the switch expression as `UselessParentheses` violation.

**Running PMD through:** CLI and Gradle
